core/services/ocr2/plugins/ocr2keeper/evm21/registry.go

```diff
@@ -647,45 +646,35 @@ func (r *EvmRegistry) doCheck(ctx context.Context, mercuryEnabled bool, keys []o
 	}
 
 	chResult <- checkResult{
-		ur: upkeepResults,
+		cr: upkeepResults,
 	}
 }
 
-func splitKey(key ocr2keepers.UpkeepKey) (*big.Int, *big.Int, error) {
-	var (
-		block *big.Int
-		id    *big.Int
-		ok    bool
-	)
-
-	parts := strings.Split(string(key), separator)
-	if len(parts) != 2 {
-		return nil, nil, fmt.Errorf("unsplittable key")
-	}
-
-	if block, ok = new(big.Int).SetString(parts[0], 10); !ok {
-		return nil, nil, fmt.Errorf("could not get block from key")
+func (r *EvmRegistry) getCheckBlockAndUpkeepId(key ocr2keepers.UpkeepPayload) (*big.Int, *big.Int, error) {
+	block, success := new(big.Int).SetString(string(key.CheckBlock), 10)
```

Comment:
I talked to Ryan and we can use the BlockNumber in the Trigger here. We shouldn't actually need `CheckBlock`.